tests/rados/test_pg_split.py

```diff
@@ -0,0 +1,87 @@
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from tests.rados.rados_test_util import create_pools, write_to_pools
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+from utility.utils import method_should_succeed, should_not_be_empty
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Automates PG split and merge test scenarios.
+    1. Create replicated pool
+    2.
+    """
+    try:
+        log.info(run.__doc__)
+        config = kw["config"]
+        rhbuild = config.get("rhbuild")
+        cephadm = CephAdmin(cluster=ceph_cluster, **config)
+        rados_obj = RadosOrchestrator(node=cephadm)
+        pool_obj = PoolFunctions(node=cephadm)
+        client_node = ceph_cluster.get_nodes(role="client")[0]
+        timeout = config.get("timeout", 10800)
+
+        log.info("Running PG split merge scenarios")
+        pool = create_pools(config, rados_obj, client_node)
+        should_not_be_empty(pool, "Failed to retrieve pool details")
+        write_to_pools(config, rados_obj, client_node)
+        # Set recover threads configurations
+        if not rhbuild.startswith("6"):
+            rados_obj.change_recover_threads(config=pool, action="set")
+        method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout)
+        pg_no = rados_obj.get_pool_property(pool=pool["pool_name"], props="pg_num")
+        if not pool["pg_num"] == pg_no:
+            log.error(
+                f"Actual pg_num {pg_no} does not match with expected pg_num {pool['pg_num']}"
+            )
+            raise Exception(
+                f"Actual pg_num {pg_no} does not match with expected pg_num {pool['pg_num']}"
+            )
+        bulk = pool_obj.get_bulk_details(pool["pool_name"])
+        if bulk:
+            log.error("Expected bulk flag should be False.")
+            raise Exception("Expected bulk flag should be False.")
+        new_bulk = pool_obj.set_bulk_flag(pool["pool_name"])
+        if not new_bulk:
+            log.error("Expected bulk flag should be True.")
+            raise Exception("Expected bulk flag should be True.")
+        method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout)
+        new_pgno = rados_obj.get_pool_property(pool=pool["pool_name"], props="pg_num")
+        if not new_pgno == "512":
```

Comment:
The count of PGs is not always hardcoded to 512. This no depends on many factors including no of pools on cluster, OSD count etc. 

Instead, he we can check if `new_pgno ` > ` pg_no `, upon bulk flag addition, removal.

If the new pg count if greater than old pg count, we can be sure of splits happening.

Reply:
made the changes.